checker/bin/infer-and-annotate.sh

```diff
@@ -0,0 +1,160 @@
+#!/bin/sh
+
+# This script runs the Checker Framework's signature inference
+# iteratively on a program, adding type annotations to the program, until the
+# .jaif files from one iteration are the same as the .jaif files from the
+# previous iteration (which means there is nothing new to infer anymore).
+
+# To use this script, the $CHECKERFRAMEWORK variable must be set to the
+# Checker Framework's directory. Also, insert-annotations-to-source must
+# be available from the $PATH.
+
+# This script receives as arguments:
+# 1. Processor's name (in any form recognized by CF's javac -processor argument).
+# 2. Classpath (target project's classpath).
+# 3. Any number of extra processor arguments to be passed to the checker.
+# 4. List of paths to .jaif files -- used as input (optional).
+# 5. List of paths to .java files in a program.
+
+# Example of usage:
+# ./infer-and-annotate.sh "LockChecker,NullnessChecker" \
+#     $JSR308/plume-lib/java/plume.jar -AprintErrorStack \
+#     `find $JSR308/plume-lib/java/src/plume/ -name "*.java"`
+
+# In case of using this script for Android projects, the classpath must include
+# paths to: android.jar, gen folder, all libs used, source code folder.
+# The Android project must be built with "ant debug" before running this script.
+
+# TODO: This script deletes all .unannotated files, including ones that could
+# have been generated previously by another means other than using this script.
+# We must decide if we want to make a backup of previously generated
+# .unannotated files, or if we want to keep the first set of generated
+# .unannotated files.
+
+# Halts the script when a nonzero value is returned from a command.
+set -e
+
+# Path to directory that will contain .jaif files after running the CF
+# with -AinferSignatures
+SIGNATURE_INFERENCE_DIR=build/signature-inference
+
+# Path that will contain .class files after running CF's javac. This dir will
+# be deleted after executing this script.
+TEMP_DIR=build/temp-signature-inference-output
+
+# Path to directory that will contain .jaif files from the previous iteration.
+PREV_ITERATION_DIR=build/prev-signature-inference
+
+# Path to annotation-file-utilities.jar
+AFU_JAR="${CHECKERFRAMEWORK}/../annotation-tools/annotation-file-utilities/annotation-file-utilities.jar"
```

Comment:
Instead or in addition to changes to the javac scripts, changes to CheckerMain/CheckerDevelMain might be necessary to add the AFU dependency. That dependency is currently only in the build and Eclipse files, but not in these scripts.